runtime/lua/vim/lsp/buf.lua

```diff
@@ -143,6 +143,79 @@ local function select_client(method, on_choice)
   end
 end
 
+--- Utility function to request formatting from attached language servers. Defaults to
+--- requesting formatting from all servers attached to the current buffer.
+---
+--- @param opts table Optional table which holds the following optional fields:
+---     - formatting_options (table):
+---         Can be used to specify FormattingOptions. Some unspecified options will be
+---         automatically derived from the current Neovim options.
+---         @see https://microsoft.github.io/language-server-protocol/specification#textDocument_formatting
+---     - timeout_ms (integer, default 1000):
+---         Time in milliseconds to block for formatting requests. Formatting requests are current
+---         synchronous to prevent editing of the buffer.
+---     - bufnr (number):
+---         Restrict formatting to the clients attached to the given buffer, defaults to the current
+---         buffer (0).
+---     - filter (function):
+---         Restrict formatting to client ids returned by this function. Takes as argument the
+---         clients attached to the buffer (default to the current buffer) and returns a list
+---         of client objects from which to request formatting.
+---
```

Comment:
```suggestion
--- @param opts table|nil Optional table which holds the following optional fields:
---     - formatting_options (table|nil):
---         Can be used to specify FormattingOptions. Some unspecified options will be
---         automatically derived from the current Neovim options.
---         @see https://microsoft.github.io/language-server-protocol/specification#textDocument_formatting
---     - timeout_ms (integer|nil, default 1000):
---         Time in milliseconds to block for formatting requests. Formatting requests are current
---         synchronous to prevent editing of the buffer.
---     - bufnr (number|nil):
---         Restrict formatting to the clients attached to the given buffer, defaults to the current
---         buffer (0).
---     - filter (function|nil):
---         Predicate to filter clients used for formatting. Received a client
---         as argument and must return a boolean.
---
```